### PR TITLE
[AIRFLOW-2772] BigQuery hook allow specifying both the partition fiel…

### PR DESCRIPTION
**Pull request by xnuinside**
…d name and table name with $

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2772
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -1662,11 +1662,8 @@ def _cleanse_time_partitioning(destination_dataset_table, time_partitioning_in):
 
     time_partitioning_out = {}
     if destination_dataset_table and '$' in destination_dataset_table:
-        if time_partitioning_in.get('field'):
-            raise ValueError(
-                "Cannot specify field partition and partition name"
-                "(dataset.table$partition) at the same time")
-        time_partitioning_out['type'] = 'DAY'
+        if not time_partitioning_in.get('field') and not time_partitioning_in.get("type"):
```
Comment (kaxil) [inline airflow/contrib/hooks/bigquery_hook.py]: Why this check?

Comment (kaxil) [inline airflow/contrib/hooks/bigquery_hook.py]: I don't think we need this check. If `$` is present in table name, it should be partitioned by `DAY` as I guess "The only type supported is DAY, which will generate one partition per day."

Comment (kaxil) [reply airflow/contrib/hooks/bigquery_hook.py]: Also update this https://github.com/apache/incubator-airflow/blob/7d60d2686d40342255a5e326f31d3cc42c72a68b/airflow/contrib/hooks/bigquery_hook.py#L566

Comment (xnuinside) [reply airflow/contrib/hooks/bigquery_hook.py]: you right, 5 sec

**Comment by xnuinside** [comment]
@kaxil, fixed

**Review by kaxil — APPROVED** [review]
LGTM

**Pull request merged**